### ISSUE TITLE
Quantity formatting: Trailing zeros

### ISSUE THREAD

**Issue by pgrawehr** [angularsen/UnitsNet#786]
The documentation of `QuantityFormatter.Format` specifies that (and it also behaves like that):
```
/// "s1", "s2", ..., "sN": The value with N significant digits after the radix followed by the unit abbreviation. For example,
/// "s4" would return "1.2345 m" if <see cref="IQuantity.Value" /> is 1.2345678. Trailing zeros are omitted.
```

Is there a reason trailing zeros are ommitted? Besides that this constantly changes the length of the output string (if the value changes slightly), it may present a value different from the expectation. For instance, if I have a temperature sensor from which I know it is accurate to 0.01 °C, I would want to use "s2" to print its value with two digits. The output now prints either "20.02 °C" or "20 °C", just because the measurement has slightly changed. Reading the later, I don't know that the measurement was actually "20.00 °C". 

For a scientist, it does make a difference whether a reading says "20 °C" or "20.00 °C". I'd like to see an option to force the number of digits to be fixed (much like the "F" default number format specifier does).

**Comment by tmilnthorp** [comment]
It becomes difficult to support all the permutations for custom format strings on a single object which holds multiple internal values. For the most flexibility, you will want to use the built-in string formatting. For your case, I would suggest something as follows:

```csharp
var temperature = UnitsNet.Temperature.FromDegreesCelsius( 20.02 );
var output = string.Format( "{0:F3} {1:a}", temperature.Value, temperature );
```

This will result in the string "20.020 °C"

**Comment by pgrawehr** [comment]
I could also make sense to allow `string.Format( "{0:F3 a}", temperature );` That would allow for maximum flexibility. Idealy, `string.Format( "{0:F3 a2}", temperature );` would then return a converted value (`72.223 °F`), since the value in celcius with another unit name is meaningless. 

Supporting all the standard format strings would be a good idea. The overlap with the "g" isn't so big of a deal I think, since "G" is still available and (unless you are into astrophysics, maybe) we usually use the unit that best suits the number, so that scientific notation is not needed.

**Comment by tmilnthorp** [comment]
I agree, we just don't support compound format strings currently. I created #788 to get us a bit closer to what you need.

FYI ```"{0:a2}"``` would not give °F for a °C value. It's only for when there multiple abbreviations for a unit. This will throw a ```FormatException```:
```csharp
var temperature = UnitsNet.Temperature.FromDegreesCelsius( 20.02 );
var output = string.Format( "{0:a2}", temperature );
```

An example use for rotational speed:
```csharp
var rotationalSpeed = RotationalSpeed.FromDegreesPerMinute( 123.456 );
var output = string.Format( "{0:a0} OR {0:a1}", rotationalSpeed );
```
Output: ```°/min OR deg/min```

**Comment by pgrawehr** [comment]
Oh, ok, thanks for clarifying that. So if I want to format the temperature in Fahrenheit, I'll have to do `temperature.ToUnit(TemperatureUnit.DegreeFahrenheit).ToString("s2"}` or similar.

**Comment by tmilnthorp** [comment]
Correct

**Comment by angularsen** [comment]
Fixed in #1047
